qml/Panel/Indicators/IndicatorsLight.qml

```diff
@@ -22,12 +23,126 @@ import Powerd 0.1
 import Lights 0.1
 import QMenuModel 0.1 as QMenuModel
 import Unity.Indicators 0.1 as Indicators
+import Wizard 0.1
 
 QtObject {
     id: root
 
     property color color: "darkgreen"
+    property int onMillisec: 1000
+    property int offMillisec: 3000
 
+    property double batteryLevel: 0
+    property string deviceState: ""
+    property bool hasMessages: false
+
+    property string batteryIconName: Status.batteryIcon
+    property string displayStatus: Powerd.status
+
+    onDisplayStatusChanged: {
+        updateLightState("onDisplayStatusChanged")
+    }
+
+    onBatteryIconNameChanged: {
+        updateLightState("onBatteryIconNameChanged")
+    }
+
+    function updateLightState(msg) {
+        console.log("updateLightState: " + msg + ", hasMessages: " + hasMessages + ", icon: " 
+            + batteryIconName + ", displayStatus: " + displayStatus + ", deviceState: " 
+            + deviceState + ", batteryLevel: " + batteryLevel)
+
+        // only show led when display is off
+        if(displayStatus == Powerd.On) {
+            Lights.state = Lights.Off
+            return
+        }
+
+        //
+        // priorities:
+        //   unread messsages (highest), full&charging, charging, low   
+        // 
+        // Icons: (see device.s from indicator-power)
+        //   %s-empty-symbolic             empty
+        //   %s-caution-charging-symbolic  charging  [ 0..10)
+        //   %s-low-charging-symbolic      charging  [10..30)
+        //   %s-good-charging-symbolic     charging  [30..60)
+        //   %s-full-charging-symbolic     charging  [60..100]
+        //   %s-full-charged-symbolic      fully charged
+        //   %s-low-symbolic               ?
+        //   %s-full-symbolic              ?
+        //
+        // device-state: (see system-settings\plugins\battery)
+        //   fully-charged
+        //   charging
+        //   discharging
+        //
+        // Indicators:
+        //   unread notifications : darkgreen pulsing (1000/3000)
+        //   charging             : white continuously
+        //   battery full         : green  continuously
+        //   battery low          : orangered pulsing (500/3000)
+        //
+        // Notes:
+        //   Icon name 'full-charged' comes late (45m after 100%)
+        //   so also check device-state and battery-level
+        //
+        //   Battery low warning dialog on screen shows up at 10%
+        //   but 'caution' icon at 9%.
+        // 
+
+        var lColor = ""
+        var lOnMS = -1
+        var lOffMS = -1
+
+        var isCharging = batteryIconName.indexOf("charging") >= 0
+        if(!isCharging)
+            isCharging = deviceState != "discharging"
+
+        if(hasMessages) { 
```

Comment:
The part of the function starting with this line would be better coded as a few `State` elements: it would be easier to read and verify for correctness. (you can read about it [here](https://doc.qt.io/qt-5/qtquick-statesanimations-states.html))

Reply:
The states might be easier to read but the if statements needed to determine the actual state will remain the same.

It looks like when using State the order of the property value changes is not specified. Due to the current implementation of the Lights plugin the led must be turned _off_ and _on_ after changing the color and pulse rates.

I don't mind using State but do you know of a way to get the led off/on triggered at the end?